wezterm-gui/src/termwindow/mod.rs

```diff
@@ -1018,44 +1018,124 @@ impl TermWindow {
+                        .collect::<Vec<_>>(),
+                );
+                pane.send_paste(
+                    paths_string.as_deref().unwrap_or(
+                        (paths
+                            .iter()
+                            .map(|path| {
+                                self.config
+                                    .quote_dropped_files
+                                    .escape(&path.to_string_lossy())
+                            })
+                            .collect::<Vec<_>>()
+                            .join(" ")
+                            + " ")
+                            .as_str(),
+                    ),
+                )?;
                 Ok(true)
             }
             WindowEvent::DraggedFile(_) => Ok(true),
         }
     }
+    fn user_dropped_event(
+        &mut self, 
+        event_name: &str, 
+        data: Vec<String>
+    ) -> Option<String> {
+        match config::run_immediate_with_lua_config(|lua| {
+            if let Some(lua) = lua {
+                let tabs = self.get_tab_information();
+                let panes = self.get_pane_information();
+                let active_tab = tabs.iter().find(|t| t.is_active).cloned();
+                let active_pane = panes.iter().find(|p| p.is_active).cloned();
+                let (modifiers, _) = self.current_modifier_and_leds;
+
+                let v = config::lua::emit_sync_callback(
+                    &lua,
+                    (
+                        event_name.to_string(),
+                        (
+                            active_tab.clone(),
+                            active_pane.clone(),
+                            modifiers.to_string(),
+                            data.clone(),
+                        ),
+                    ),
+                )?;
+                match &v {
+                    mlua::Value::Nil => Ok(None),
+                    mlua::Value::Table(t) => {
+                        let mut result_string = String::new();
+                        for i in 1..=t.len()? {
+                            if let Some(v) = t.get(i)? {
+                                if !result_string.is_empty() {
+                                    result_string.push(' ');
+                                }
+                                result_string.push_str(&String::from_lua(v, &lua)?);
+                            }
+                        }
+                        if result_string.is_empty() {
+                            return Ok(None);
+                        }
+                        Ok(Some(result_string))
+                    }
+                    mlua::Value::String(s) => {
+                        let v = s.to_str()?;
+                        if v.is_empty() {
+                            return Ok(Some("".to_string()));
+                        }
+                        Ok(Some(v.to_string()))
+                    }
+                    _ => Ok(Some(String::from_lua(v, &lua)?)),
```

Comment:
let lua implicitly `tostring` anything that isn't Nil: 

```suggestion
                    value => match value.to_string() {
                            Ok(s) => return Some(s),
                            Err(err) => {
                               log::warn!("{event_name}: error converting result to string: {err:#}");
                               return None
                            }
                        }
```